recipes/fmt/all/conanfile.py

```diff
@@ -64,65 +58,63 @@ def config_options(self):
             self.options.with_os_api = False
 
     def configure(self):
-        if self.options.header_only:
-            del self.options.fPIC
-            del self.options.shared
-            del self.options.with_os_api
-        elif self.options.shared:
-            del self.options.fPIC
-
-    def validate(self):
-        if self.options.get_safe("shared") and self._is_msvc and "MT" in msvc_runtime_flag(self):
-            raise ConanInvalidConfiguration(
-                "Visual Studio build for shared library with MT runtime is not supported"
-            )
+        try:
+            if self.options.header_only:
+                del self.options.fPIC
+                del self.options.shared
+                del self.options.with_os_api
+            elif self.options.shared:
+                del self.options.fPIC
+        except Exception:
+            pass
 
     def package_id(self):
-        if self.options.header_only:
+        if self.info.options.header_only:
```

Comment:
Maybe we could add a comment here, because from Conan 1.50 this will change to be `self.info.clear()` instead `self.info.header_only()`